### DISCUSSION TITLE
Non-tissue area is considered as part of tissue

### DISCUSSION THREAD

**Discussion by agathalactiae** [giotto-suite/Giotto#263] [Q&A] [answered]
Hi everyone, 

Currently I am working with data from 10X Visium (https://www.10xgenomics.com/resources/datasets/adult-mouse-brain-ffpe-1-standard-1-3-0), yet I have a problem when creating the Giotto object as some of the non-tissue area is also assigned as tissue on my Giotto object. 

This is the code I use to create the object:

`giotto_object = createGiottoVisiumObject(visium_dir="~/FFPE brain/spatial", h5_visium_path= "~/FFPE brain/spatial/Visium_FFPE_Mouse_Brain_raw_feature_bc_matrix.h5", h5_tissue_positions_path="/Users/agathanabillalestari/FFPE brain/spatial/spatial/tissue_positions_list.csv", instructions = the_instructions,h5_image_png_path = "/Users/agathanabillalestari/FFPE brain/spatial/spatial/tissue_lowres_image.png" , h5_json_scalefactors_path = "/Users/agathanabillalestari/FFPE brain/spatial/spatial/scalefactors_json.json")`

but instead of having the dots populated on the tissue image only, I got them in all over the square

<img width="1122" alt="Screenshot 2022-06-16 at 13 39 54" src="https://user-images.githubusercontent.com/93100652/174072254-ae26b702-ee7a-410e-8e09-48e9c2877363.png">

<img width="1440" alt="Screenshot 2022-06-16 at 13 35 00" src="https://user-images.githubusercontent.com/93100652/174071685-8844b745-853c-4c7a-a0ef-1b9b2e24917f.png">

I tried to process the data with Seurat and make it as Seurat object, and there is no issue on it. Thus, I wonder if I did something wrong in generating the Giotto object. This is a problem, because the non-tissue area is also processed as transcription and grouped as a distinct cluster 

<img width="1114" alt="Screenshot 2022-06-16 at 13 41 46" src="https://user-images.githubusercontent.com/93100652/174072140-d631c5e1-0fb1-4bbf-814c-75ed3213e4d5.png">

Would anyone please suggest me the correction that I should do to solve this issue?

Thank you and best regards, 

Agatha

**Comment by agathalactiae** [comment]
Hi everyone, 

sorry that I finally solved it. It turns out that I should use this code when generating the spatplot2D, and after that subsetting the data that has only one in it.


`spatPlot2D(gobject=giotto_object, show_image = TRUE, point_alpha = 0.75, coord_fix_ratio = 1, cell_color = 'in_tissue', save_param = list(save_name = "spatplot2D", base_width =6, base_height = 6, save_format = "pdf"))

pDataDT(giotto_object)
spatPlot(gobject = giotto_object, cell_color = 'in_tissue', point_size = 2, cell_color_code = c('0' = 'lightgrey', '1' = 'blue'), save_param = list(save_name = 'tryin_tissue'))
metadata = pDataDT(giotto_object)
in_tissue_barcodes = metadata[in_tissue == 1]$cell_ID
giotto_object = subsetGiotto(giotto_object, cell_ids = in_tissue_barcodes)
`

**Comment by RubD** [comment, marked as answer]
Hi @curiegat 
Yes, you're right. You need to subset the data first and only include the spots that cover tissue. This information is encoded in the `in_tissue` column in the metadata.
